exonum-java-binding/core/rust/src/storage/db.rs

```diff
@@ -128,6 +128,32 @@ impl View {
         }
     }
 
+    /// Creates checkpoint for the owned Fork instance.
+    ///
+    /// Panics if it is not possible (`View::can_rollback` returns false).
+    pub fn create_checkpoint(&mut self) {
+        match self {
+            View::Owned(ViewOwned::Fork(fork)) => fork.flush(),
+            View::RefMutFork(fork) => fork.flush(),
+            _ => panic!("Cannot create checkpoint because this View does not support it"),
+        }
+    }
+
+    /// Rollbacks owned Fork to the latest checkpoint.
+    /// If no checkpoint was created (`create_checkpoint` method was never called),
+    /// rollbacks all changes in Fork.
+    ///
+    /// Does not affect database, but only a specific Fork instance.
+    ///
+    /// Panics if it is not possible (`View::can_rollback` returns false).
+    pub fn rollback(&mut self) {
+        match self {
+            View::Owned(ViewOwned::Fork(fork)) => fork.rollback(),
+            View::RefMutFork(fork) => fork.rollback(),
+            _ => panic!("Cannot rollback because this View does not support it"),
```

Comment:
Can we include the actual enum variant type?

Reply:
Done, using `Debug` implementation. Also did the same change for `into_fork` method.